pkgs/apis/v1/issuers.go

```diff
@@ -0,0 +1,87 @@
+package v1
+
+import (
+	issuerv1alpha1 "github.com/cert-manager/issuer-lib/api/v1alpha1"
+)
+
+type AuthType int
+
+const (
+	AuthTypeServiceKey AuthType = 1
+	AuthTypeAPIToken   AuthType = 2
+)
+
+var _ issuerv1alpha1.Issuer = (*OriginIssuer)(nil)
+var _ issuerv1alpha1.Issuer = (*ClusterOriginIssuer)(nil)
+
+func (iss *OriginIssuer) GetStatus() *issuerv1alpha1.IssuerStatus {
+	return &iss.Status
+}
+
+func (iss *OriginIssuer) GetIssuerTypeIdentifier() string {
+	return "originissuers." + GroupVersion.Group
+}
+
+func (iss *OriginIssuer) GetAuthSecretNamespace(_ string) string {
+	return iss.Namespace
+}
+
+func (iss *OriginIssuer) GetAuthSecretKeySelector() *SecretKeySelector {
+	switch {
+	case iss.Spec.Auth.ServiceKeyRef != nil:
+		return iss.Spec.Auth.ServiceKeyRef
+	case iss.Spec.Auth.TokenRef != nil:
+		return iss.Spec.Auth.TokenRef
+	}
+	return nil
+}
+
+func (iss *OriginIssuer) GetAuthType() AuthType {
+	switch {
+	case iss.Spec.Auth.ServiceKeyRef != nil:
+		return AuthTypeServiceKey
+	case iss.Spec.Auth.TokenRef != nil:
+		return AuthTypeAPIToken
+	}
+	return 0
+}
+
+func (iss *OriginIssuer) GetRequestType() RequestType {
+	return iss.Spec.RequestType
+}
+
+func (iss *ClusterOriginIssuer) GetStatus() *issuerv1alpha1.IssuerStatus {
+	return &iss.Status
+}
+
+func (iss *ClusterOriginIssuer) GetIssuerTypeIdentifier() string {
+	return "clusteroriginissuers." + GroupVersion.Group
+}
+
+func (iss *ClusterOriginIssuer) GetAuthSecretNamespace(clusterResourceNamespace string) string {
+	return clusterResourceNamespace
+}
+
+func (iss *ClusterOriginIssuer) GetAuthSecretKeySelector() *SecretKeySelector {
```

Comment:
nit: Since `ClusterOriginIssuer` and `OriginIssuer` use the same `OriginIssuerAuthentication` type, you could remove a small amount of duplication by having common helper functions to implement `GetAuthSecretKeySelector` and `GetAuthType` (possibly even making them methods of `OriginIssuerAuthentication`).